source/tutorials/nix-shell-in-shebang.md

```diff
@@ -0,0 +1,76 @@
+# Reproducible interpreted scripts
+
+In this tutorial, you will learn how to use Nix to create and run reproducible interpreted scripts with [shebang].
+
+## Requirements
+
+- A working [Nix installation](install-nix)
+- Familiarity with [Bash]
+
+## A trivial script with non-trivial dependencies
+
+Take the following script, which fetches the content XML of a URL, converts it to JSON, and formats it for better readability:
+
+```bash
+#! /bin/bash
+
+curl https://github.com/NixOS/nixpkgs/releases.atom | xml2json | jq .
+```
+
+It requires the programs `curl`, `xml2json`, and `jq`.
+It also requires the `bash` interpreter.
+If any of these dependencies are not present on the system running the script, it will fail partially or altogether.
+
+With Nix, we can declare all dependencies explicitly, and produce a script that will always run on any machine that supports Nix and the required packages taken from Nixpkgs.
+
+## The script
+
+A [shebang] is the first line of a script starting with `#!`.
+It determines which program to use for running the script.
+
+[Bash]: https://www.gnu.org/software/bash/
+[shebang]: https://en.wikipedia.org/wiki/Shebang_(Unix)
+
+We will use the shebang line `#! /usr/bin/env nix-shell`.
+
+`/usr/bin/env` is a program available on most modern Unix-like operating systems.
+It takes a command name as argument and will run the first executable by that name it finds in the directories listed in the environment variable `$PATH`.
+
+We use [`nix-shell` as a shebang interpreter].
+It takes the following parameters relevant for our use case:
+
+[`nix-shell` as a shebang interpreter]: https://nixos.org/manual/nix/stable/command-ref/nix-shell.html#use-as-a--interpreter
+- `-i` tells which program to use for interpreting the rest of the file
+- `-p` lists packages that should be present in the interpreter's environment
+- `-I` explicitly sets the search path for packages
+- `--pure` clears the environment before start - more details in the [nix-shell man page]: https://nixos.org/manual/nix/stable/command-ref/nix-shell.html#options
+
+Create a file named `nixpkgs-releases.sh` with the following content:
+
+```shell
+#!/usr/bin/env nix-shell 
+#! nix-shell -i bash --pure
+#! nix-shell -p cacert curl jq python3Packages.xmljson
```

Comment:
Please also include `-p bash`; while implicit in this case (it is included by something else), this omission in the prior tutorial led to me wasting a fair chunk of my time! See also: https://discourse.nixos.org/t/nix-shell-shebang-default-nix-no-such-file-or-directory/24246/2?u=n8henrie

EDIT: (Adding `-p bash` was the main reason I was tagged to contribute to this PR :) thanks for picking up and running with it!)

Reply:
Thank you for the feedback Nathan, this is included now.